### ISSUE TITLE
Smart image classification

### ISSUE THREAD

**Issue by deckerst** [deckerst/aves#66]
### Discussed in https://github.com/deckerst/aves/discussions/60

<div type='discussions-op-text'>

<sup>Originally posted by **quantified1995** August 31, 2021</sup>
Didn't want to add this to my previous post so I figured I'd post a separate (forgive me as I'm new to GitHub and app development)

Another incredible feature I would love to see in this app would be smart image classification. I've seen quite a few apps implement this and I consider it a make or break feature. I'm not sure to what extent the difficulty would be to implement this but it would be awesome to see. </div>

**Comment by Subbarao6338** [comment]
Image classification based on people, animals, location, scenary etc

**Comment by flexagoon** [comment]
The [tflite_flutter](https://pub.dev/packages/tflite_flutter) package has an example app implementing image classification

**Comment by DodoLeDev** [comment]
I can do some mockups about how this feature could be integrated into the app! 😋

**Comment by OkyDooky** [comment]
I was even thinking of a very simple one that could use keywords and such in the filename or other parts of the metadata to suggest tags. The app I download most of my images from (that I wish to tag in Aves) adds the artist name and the title of the work in the filename automatically. So, if I download something that mentions a character, like Zelda, it could recommend the top tags I use on other images with "Zelda" in the title, lile "Link" or "The Legend of Zelda" and so on.
This might be an easy stepping stone towards that, perhaps.

**Comment by Inky-developer** [comment]
Maybe this could be useful: https://github.com/slavabarkov/tidy

**Comment by deckerst** [comment]
Tidy seems cool but it's huge, likely because of the model. Having the model published as part of Aves is out, but it could be installed as a separate APK.

In any case, thanks for pointing it out!

**Comment by devSJR** [comment]
@deckerst It would really be great to have this in Aves. TIDY works pretty well, so I guess the underlying technology is reliable. But it seems that nobody is working any longer on TIDY.

**Comment by ElmerJeanpierreLopez** [comment]
Good day.

AVES was the solution that had in mind when I discover that Samsung Gallery tags doesn't apearse as metadata. On this, I want to suggest a feature, suggest - tags based on the existing tags. I see that some others galleries (Samsung too) has some general classes,  but I think I'd would be a better focus to use meaningful categories for each user (tags). I understand that will be difficult because involves train a model on each user data probably but I would be great.   

Thank you for the work

**Comment by Heavy-A** [comment]
This would be great to include in a workflow to go through all untagged /minimally tagged files and suggest (ask, accept, no_accept) tags. Suggesting tags already in the DB, otherwise suggest new ones.

**Comment by Jakarrrg** [comment]
There is an open source app on f-droid that utilizes local ai model to do that. Maybe it could help: https://github.com/smartscanapp/smartscan-android

Also the "Box" app (https://github.com/jegly/Box) recently featured an ai tool that "tags" any photo. The model they use could be utilized in Aves too.

**Comment by devSJR** [comment]
That's right, there are still more FLOSS apps to address this topic. Here is my current recommendation. 

https://github.com/IacobIonut01/ReFra
